File: sem3/map/ToyLanguageInterpreter/src/view/commands/ExecuteAllStepsCommand.java
package view.commands;

import controller.Controller;
import utils.exceptions.InterpreterException;

import java.util.Scanner;

/**
 * Created by mirko on 14/11/2016.
 */
public class ExecuteAllStepsCommand extends Command {
    private Controller controller;

    public ExecuteAllStepsCommand(String key, String description, Controller controller) {
        super(key, description);
        this.controller = controller;
    }

    @Override
    public void execute(Scanner scanner) {
        try {
            controller.executeAllSteps();
            System.out.println(controller.currentProgramStatesToString());
            System.out.flush();
        } catch (InterpreterException error) {
            System.out.println(error.toString());
            System.out.flush();
        }
    }
}

File: sem3/map/ToySharpLanguageInterpreter/ToySharpLanguageInterpreter/src/view/commands/ExecuteAllStepsCommand.cs
using System;
using System.IO;

namespace ToySharpLanguageInterpreter
{
	public class ExecuteAllStepsCommand : Command
	{
		private Controller controller;

		public ExecuteAllStepsCommand(string key, string description, Controller controller) : base(key, description)
		{
			this.controller = controller;
		}


		public override void execute()
		{
			try
			{
				controller.executeAllSteps();
				Console.WriteLine(controller.currentProgramToString());
			}
			catch (InterpreterException error)
			{
				Console.WriteLine(error.Message);
			}
		}
	}

}